### ISSUE TITLE
[Bug] World crash due to autocrafting with liquids

### ISSUE THREAD

**Issue by MACSkeptic** [refinedmods/refinedstorage#3157]
Ran into exactly this issue when playing sky factory one this week (see related https://github.com/DarkPacks/SkyFactory-One/issues/149): https://github.com/refinedmods/refinedstorage/blob/9c21cad9f62fdfe022d3a6049ce3f9b9a112c505/src/main/java/com/refinedmods/refinedstorage/apiimpl/autocrafting/task/v6/node/NodeRequirements.java#L140

My setup:

- Crafter with recipe for blaze rod: 0.2 buckets of blazing blood -> blaze rod, pointing at a mob grinding utils singularty tank
- External storage pointing at a mob grinding utils jumbo tank with blazing blood (plenty, dozens of buckets of content)

Request to craft a blaze rod and boom, world crashes.

[crash-2021-11-30_21.23.12-server.txt](https://github.com/refinedmods/refinedstorage/files/7628741/crash-2021-11-30_21.23.12-server.txt)

-------------

The only way of loading my world again was to edit the refined storage .dat files and change my controller redstone mode (from ignore to only work on redstone on).

This allowed me to load the world again and I verified that turning the refined storage system on causes a crash.

Later I just removed this blaze rod crafting setup (it was the _only_ change I made) and verified that I could turn my refined storage system back up again with no issues, thus determining the exact culprit.

**Comment by MACSkeptic** [comment]
![storage](https://user-images.githubusercontent.com/53897/144126907-eb9ef2c8-df64-48db-9edd-896cf51c9880.png)
![setup](https://user-images.githubusercontent.com/53897/144126913-39e65951-324c-4dea-905f-679e04a21cd5.png)

**Comment by gtokarz** [comment]
I think I have reproduction scenario for this one if it helps:
I was using mekanism tank for filling buckets, by accident I have set up pattern as follows:
Input:
2000mb of water, 1 bucket, output 1 water bucket, additionally I have decided to use exporter instead of specyfing water amount, but forgot to remove it. When 64 buckets were requested it did like 10 or so and froze, nothing could be crafted again in that tank (mostly to it being full), when I have placed buckets in manually, it did couple of them and crashed with same error as above.

**Comment by Spongman** [comment]
I'm seeing the same thing. now my world won't load :(

```
java.lang.IllegalStateException: Bad!
	at com.refinedmods.refinedstorage.apiimpl.autocrafting.task.v6.node.NodeRequirements.getSingleFluidRequirementSet(NodeRequirements.java:140)
	at com.refinedmods.refinedstorage.apiimpl.autocrafting.task.v6.node.ProcessingNode.update(ProcessingNode.java:123)
	at com.refinedmods.refinedstorage.apiimpl.autocrafting.task.v6.CraftingTask.update(CraftingTask.java:191)
	at com.refinedmods.refinedstorage.apiimpl.autocrafting.CraftingManager.updateTasks(CraftingManager.java:170)
	at com.refinedmods.refinedstorage.apiimpl.autocrafting.CraftingManager.update(CraftingManager.java:134)
	at com.refinedmods.refinedstorage.apiimpl.network.Network.update(Network.java:165)
	at com.refinedmods.refinedstorage.apiimpl.network.NetworkListener.onWorldTick(NetworkListener.java:17)
	at net.minecraftforge.eventbus.ASMEventHandler_1082_NetworkListener_onWorldTick_WorldTickEvent.invoke(.dynamic)
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at net.minecraftforge.fml.hooks.BasicEventHooks.onPostWorldTick(BasicEventHooks.java:100)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:857)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:78)
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642)
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Spongman** [comment]
Ok, this is pretty frustrating. I re-built the mod with those exceptions commented out and my world loads just fine, and it seems to work fine, too.
So, as far as I can tell those exceptions serve a single purpose: to make a player's world unloadable.

**Comment by gtokarz** [comment]
@Spongman there is a way to restore your world I had same problem edit file saves/<save>/data/refinedstorage_networks.dat in NBT editor. Change redstone mode of your network to something else (1/2 depending to what you had in your case) 
![image](https://user-images.githubusercontent.com/12443385/149623552-9acc9c61-3411-4d99-8942-b4cbf94940b6.png)
It will make sure your controller will fire a network only on redstone, remove pattern causing crash (if you can recall what you did before crash)

**Comment by Darkere** [comment]
This is fixed in 1.9.17
